### PR TITLE
Replace Mocha & Istanbul with Jest & Codecov

### LINKED ISSUE #94: Replace Mocha & Istanbul with Jest & Codecov
[Jest](https://jestjs.io/) will replace Mocha as this project's test runner. We should also be able to remove Istanbul (the NPM package called `nyc`) and replace it with [Codecov](https://codecov.io/).

Integrating Codecov will be handled by @WesCossick; everything else will be handled by @pbelbin.

### PR DESCRIPTION
**Pull request by pbelbin**
closes #94

### CODE REVIEW

**Comment by pbelbin** [comment]
Here is how the output from 'npm run test' looks:

```
Peters-MBP:waterfall-cli peterbelbin$ npm run test

> waterfall-cli@0.2.0 test /Users/peterbelbin/github/waterfall-cli
> jest

 PASS  test/pizza-ordering.test.js
  Pizza ordering
    Built-in abilities
      ✓ Displays version (80ms)
      ✓ Displays version for a command (66ms)
      ✓ Displays help screen (66ms)
      ✓ Displays help screen for a command (64ms)
      ✓ Displays help screen when no arguments (62ms)

 PASS  test/utils.test.js
  Utils
    #processArguments()
      ✓ Handles normal arguments (1ms)
      ✓ Handles data with special characters (1ms)
      ✓ Handles dangling equals sign
    #retrieveAppInformation()
      ✓ Retrieves app info (1ms)
      ✓ Cannot find package.json
    #getMergedSpec()
      ✓ Gets top-level spec (1ms)
      ✓ Gets merged spec (1ms)
      ✓ Complains about multiple .js files (2ms)
      ✓ Complains about multiple .json files
      ✓ Complains about bad JSON (1ms)
    #organizeArguments()
      ✓ Handles no arguments
      ✓ Handles simple command (1ms)
      ✓ Handles simple command with data (31ms)
      ✓ Handles simple command with multi-word data (1ms)
      ✓ Handles simple command with flag after data (2ms)
      ✓ Ignores flag in data (3ms)
      ✓ Ignores options in data (3ms)
      ✓ Handles simple command with longer data (2ms)
      ✓ Handles data with special characters (2ms)
      ✓ Handles integer data (2ms)
      ✓ Handles float data (8ms)
      ✓ Handles multiple commands (1ms)
      ✓ Handles flag (1ms)
      ✓ Handles cascading flag (1ms)
      ✓ Handles shorthand flag (1ms)
      ✓ Handles multiple flags (1ms)
      ✓ Handles cascading flag before command (1ms)
      ✓ Handles integer option (1ms)
      ✓ Handles float option (3ms)
      ✓ Handles cascading option (1ms)
      ✓ Handles shorthand option (1ms)
      ✓ Treats bad command as data (3ms)
      ✓ Complains about missing option value (1ms)
      ✓ Complains about unrecognized option value type (1ms)
      ✓ Complains about unrecognized data type (2ms)
      ✓ Complains about unrecognized flag (1ms)
      ✓ Complains about non-permitted data (5ms)
      ✓ Complains about option value not in "values" (1ms)
      ✓ Complains about data not in "values" (1ms)
      ✓ Complains about option value not being integer (1ms)
      ✓ Complains about option value not being float (3ms)
      ✓ Complains about data not being integer (4ms)
      ✓ Complains about data not being float (7ms)
    #constructInputObject()
      ✓ Handles combination of input (3ms)
      ✓ Complains about missing required option (2ms)
      ✓ Complains about missing required data (1ms)
    #getAllProgramCommands()
      ✓ Gets all commands (1ms)
    #convertDashesToCamelCase()
      ✓ Normal string
      ✓ With numbers (1ms)
    #files
      ✓ Detects directory is directory
      ✓ Detects file is not directory
      ✓ Detects file is file
      ✓ Detects directory is not file
      ✓ Retrieves first level files (1ms)
      ✓ Retrieves first level directories
      ✓ Retrieves all directories
      ✓ GetFiles() returns empty array if path not found (1ms)
      ✓ GetDirectories() returns empty array if path not found
      ✓ GetAllDirectories() returns empty array if path not found

 PASS  test/screens.test.js
  Screens
    #version()
      ✓ Full version screen (1ms)
      ✓ Missing name
    #help()
      ✓ Description line (4ms)
      ✓ Usage line (commands + flags + options) (2ms)
      ✓ Usage line (flags + options + data) (1ms)
      ✓ Flags - header (1ms)
      ✓ Flags - non-cascading (3ms)
      ✓ Flags - cascading (2ms)
      ✓ Flags - shorthand (4ms)
      ✓ Flags - description (2ms)
      ✓ Options - header (1ms)
      ✓ Options - cascading (2ms)
      ✓ Options - shorthand (2ms)
      ✓ Options - description (1ms)
      ✓ Options - description (type) (1ms)
      ✓ Options - description (required + accepts) (1ms)
      ✓ Commands - header (3ms)
      ✓ Commands - description (1ms)
      ✓ Data - header (3ms)
      ✓ Data - description (1ms)
      ✓ Data - description (required + accepts) (1ms)

------------------------|----------|----------|----------|----------|-------------------|
File                    |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------------------|----------|----------|----------|----------|-------------------|
All files               |    97.72 |    90.67 |    97.87 |    97.64 |                   |
 default-settings.js    |      100 |      100 |      100 |      100 |                   |
 error-without-stack.js |      100 |      100 |      100 |      100 |                   |
 screens.js             |    98.88 |       80 |      100 |    98.84 |               239 |
 utils.js               |    97.17 |    95.49 |    97.22 |    97.07 |... 11,413,545,546 |
------------------------|----------|----------|----------|----------|-------------------|
Test Suites: 3 passed, 3 total
Tests:       85 passed, 85 total
Snapshots:   0 total
Time:        1.822s
Ran all test suites.
Peters-MBP:waterfall-cli peterbelbin$
```

File: package.json [hunk 1]
```diff
@@ -33,11 +33,20 @@
 		"semver": "^6.3.0"
 	},
 	"devDependencies": {
+		"codecov": "^3.6.1",
```
Comment (WesCossick) [inline package.json]: No changes need to be introduced for Codecov at this point. I'll handle that.

Comment (pbelbin) [reply package.json]: Ah.  ok.

File: package.json [hunk 2]
```diff
@@ -36,8 +36,16 @@
...
 		"eslint-plugin-import": "^2.18.2",
-		"mocha": "^6.2.0",
-		"nyc": "^14.1.1",
+		"jest": "^24.9.0",
 		"prettier": "1.18.2"
+	},
+	"jest": {
+		"verbose": true,
+		"collectCoverage": true,
+		"expand": true,
```
Comment (WesCossick) [inline package.json]: Why did you use this option?

File: package.json [hunk 3]
```diff
@@ -36,8 +36,16 @@
...
+		"roots": [
```
Comment (WesCossick) [inline package.json]: The `roots` key seems unnecessary too. Removing it does not impact the tests, as far as I can tell.

**Pull request merged**